src/geometry/manifold/algebra/left_invariant_derivation.lean

```diff
@@ -0,0 +1,186 @@
+  left_invariant', pi.add_apply]⟩ }
+
+instance : has_neg (left_invariant_derivation I G) :=
+{ neg := λ X, ⟨-X, λ f g, by simp only [linear_map.map_neg, derivation.coe_neg, left_invariant',
+  pi.neg_apply]⟩ }
+
+instance : has_sub (left_invariant_derivation I G) :=
+{ sub := λ X Y, ⟨X - Y, λ f g, by simp only [linear_map.map_sub, derivation.coe_sub,
+  left_invariant', pi.sub_apply]⟩ }
+
+@[simp] lemma coe_add : ⇑(X + Y) = X + Y := rfl
+@[simp] lemma coe_zero : ⇑(0 : left_invariant_derivation I G) = 0 := rfl
+@[simp] lemma coe_neg : ⇑(-X) = -X := rfl
+@[simp] lemma coe_sub : ⇑(X - Y) = X - Y := rfl
+@[simp, norm_cast] lemma lift_add :
+  (↑(X + Y) : derivation 𝕜 C^∞⟮I, G; 𝕜⟯ C^∞⟮I, G; 𝕜⟯) = X + Y := rfl
+@[simp, norm_cast] lemma lift_zero :
+  (↑(0 : left_invariant_derivation I G) : derivation 𝕜 C^∞⟮I, G; 𝕜⟯ C^∞⟮I, G; 𝕜⟯) = 0 := rfl
+@[simp] lemma map_add : X (f + f') = X f + X f' := derivation.map_add X f f'
+@[simp] lemma map_zero : X 0 = 0 := derivation.map_zero X
+@[simp] lemma map_neg : X (-f) = -X f := derivation.map_neg X f
+@[simp] lemma map_sub : X (f - f') = X f - X f' := derivation.map_sub X f f'
+
+instance : add_comm_group (left_invariant_derivation I G) :=
+coe_injective.add_comm_group _ coe_zero coe_add coe_neg coe_sub
+
+instance : has_scalar 𝕜 (left_invariant_derivation I G) :=
+{ smul := λ r X, ⟨r • X, λ f g, by { simp only [derivation.Rsmul_apply, algebra.id.smul_eq_mul,
+            mul_eq_mul_left_iff, linear_map.map_smul, left_invariant'], left, refl }⟩ }
+
+variables (r X)
+
+@[simp] lemma coe_smul : ⇑(r • X) = r • X := rfl
+@[simp] lemma map_smul : X (r • f) = r • X f := derivation.map_smul X r f
+@[simp] lemma leibniz : X (f * f') = f • X f' + f' • X f := X.leibniz' _ _
+@[simp] lemma lift_smul (k : 𝕜) : (↑(k • X) : derivation 𝕜 C^∞⟮I, G; 𝕜⟯ C^∞⟮I, G; 𝕜⟯) = k • X := rfl
+
+variables (I G)
+
+/-- The coercion to function is a monoid homomorphism. -/
+def coe_fn_add_monoid_hom : (left_invariant_derivation I G) →+ _ :=
+⟨λ X, X.to_derivation.to_fun, coe_zero, coe_add⟩
+
+variables {I G}
+
+instance : module 𝕜 (left_invariant_derivation I G) :=
+coe_injective.module _ (coe_fn_add_monoid_hom I G) coe_smul
+
+/-- Evaluation at a point for left invariant derivation. Same thing as for generic global
+derivations (`derivation.eval_at`). -/
+def eval_at : (left_invariant_derivation I G) →ₗ[𝕜] (point_derivation I g) :=
+{ to_fun := λ X, derivation.eval_at g ↑X,
+  map_add' := λ X Y, rfl,
+  map_smul' := λ k X, rfl }
+
+lemma eval_at_apply : eval_at g X f = (X f) g := rfl
+
+@[simp] lemma eval_at_coe : derivation.eval_at g ↑X = eval_at g X := rfl
+
+lemma left_invariant : (𝒅(𝑳 I g)) (1 : G) (eval_at (1 : G) X) f = eval_at g X f :=
+(X.left_invariant'' f g)
+
+lemma eval_at_mul : eval_at (g * h) X f = (𝒅(𝑳 I g)) h (eval_at h X) f :=
+by rw [←left_invariant, L_mul, fdifferential_comp, apply_fdifferential, linear_map.comp_apply,
+  apply_fdifferential, left_invariant]
+
+lemma comp_L : (X f).comp (𝑳 I g) = X (f.comp (𝑳 I g)) :=
```

Comment:
```suggestion
lemma comp_smooth_left_mul : (X f).comp (𝑳 I g) = X (f.comp (𝑳 I g)) :=
```
or
```suggestion
lemma comp_«𝑳» : (X f).comp (𝑳 I g) = X (f.comp (𝑳 I g)) :=
```
I think using a different `L` in the lemma name and notation is confusing.

Reply:
Oh, I see there's already precedent for this. In that case, can you update the docstring for `smooth_left_mul` to mention that lemmas about it use `L` instead of `𝑳` in their names?